controllers/storagecluster/quickstarts.go

```diff
@@ -6,13 +6,24 @@ import (
 	"github.com/ghodss/yaml"
 	consolev1 "github.com/openshift/api/console/v1"
 	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 type ocsQuickStarts struct{}
 
 func (obj *ocsQuickStarts) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	qscrd := extv1.CustomResourceDefinition{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "consolequickstarts.console.openshift.io", Namespace: ""}, &qscrd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("No custom resource definition found for consolequickstart. Skipping quickstart initialiazation")
+			return nil
+		}
+		r.Log.Error(err, "Failed to fetch consolequickstart CRD")
+		return nil
```

Comment:
If it failed to fetch, shouldn't we try to fetch it again and try to deploy?

Reply:
If we have a non-IsNotFound error in a client Get() call, this is probably indicative of a larger problem. I would imagine we would want to fail and start over as we do elsewhere.

Reply:
returning an err here. it should fail the reconcile and retry again